internal/manifests/targetallocator/servicemonitor.go

```diff
@@ -0,0 +1,70 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package targetallocator
+
+import (
+	"fmt"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/internal/manifests"
+	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
+)
+
+// ServiceMonitor returns the service monitor for the given instance.
+func ServiceMonitor(params manifests.Params) (*monitoringv1.ServiceMonitor, error) {
+	if !params.OtelCol.Spec.TargetAllocator.Observability.Metrics.EnableMetrics {
+		params.Log.V(2).Info("Metrics disabled for this OTEL Collector",
+			"params.OtelCol.name", params.OtelCol.Name,
+			"params.OtelCol.namespace", params.OtelCol.Namespace,
+		)
+		return nil, nil
+	}
+
```

Comment:
Please add a condition to avoid creating sm when running in sidecar mode.

Reply:
Will target allocator support sidecar mode? It seems like setting `mode != statefulset` with target allocator enabled will be denied by the admission webhook and disabling target allocator will not generate any ta manifest either.

Reply:
We already check for this during validation, we shouldn't reproduce this check here as well imo. If anything, I'd move the check for `EnableMetrics` here from `targetallocator.go`.

Reply:
It seems like I also added checks in `targetallocator.go` , so I'll remove it from the `servicemonitor.go`, however I would like to add tests for building targetallocator and it seems like `builder_test.go`  only covers collector and OpAMPBridge, should there be dedicated test for targetallocator, or should I just append test case to the collector tests?

Reply:
If you want to add those tests, please do so in a separate test function.